sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableBuilder.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog;
+
+import com.google.common.collect.Maps;
+import org.apache.spark.sql.catalyst.analysis.NoSuchNamespaceException;
+import org.apache.spark.sql.catalyst.analysis.TableAlreadyExistsException;
+import org.apache.spark.sql.connector.expressions.Transform;
+
+import java.util.Map;
+
+/**
+ * Builder for doing table creations. Spark gets the builder from the {@link TableCatalog},
+ * calls the `withXXX` methods to set up the fields and calls the `create` method.
+ */
+public class TableBuilder {
+  protected final TableCatalog catalog;
+
+  protected final Identifier identifier;
+
+  protected final Column[] columns;
+
+  protected final Map<String, String> properties = Maps.newHashMap();
+
+  protected Transform[] partitions = new Transform[0];
```

Comment:
I think we just need to use null here.

Reply:
We need to keep the existing behavior, which passes an empty array for unpartitioned tables. Please see the above review comment.

Reply:
OK. I'm missing some comments.